### PR TITLE
Project Beats: localize blockly text

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
This adds localized strings for all blockly-related text in Project Beats (block names, toolbox categories, block tooltips, dropdown fields, etc). I've prefixed all of these keys with "blockly_" for easier identification if/when we transfer these strings into levelbuilder.

I also renamed the musicLocale import to `musicI18n` as per guidance from the platform team.

## Links

https://codedotorg.atlassian.net/browse/SL-630

## Testing story

Tested locally with en_us and another language (with stubbed translations) to ensure that all strings were correctly localized.

### CODE REVIEW

File: apps/i18n/music/en_us.json [hunk 1]
```diff
@@ -1,4 +1,53 @@
 {
+  "blockly_blockNewTrackAtStart": "new track",
```
Comment (breville) [inline apps/i18n/music/en_us.json]: Do we need to localize these non-`simple2` models?

File: apps/src/music/blockly/blocks/control.js [hunk 2]
```diff
@@ -1,57 +1,9 @@
...
-    //ctx.getFieldValue('code') +
-    Blockly.JavaScript.statementToCode(ctx, 'code') +
-    '\n}\n',
-};
-
 export const forLoop = {
   definition: {
     type: BlockTypes.FOR_LOOP,
-    message0: 'for %1 from %2 to %3 by %4',
+    message0: '%{BKY_CONTROLS_FOR_TITLE}',
```
Comment (breville) [inline apps/src/music/blockly/blocks/control.js:6]: Is this in scope without being explicitly imported?

File: apps/src/music/blockly/blocks/events.js [hunk 3]
```diff
@@ -1,14 +1,15 @@
+import musicI18n from '../../locale';
```
Comment (breville) [inline apps/src/music/blockly/blocks/events.js]: ultra nit: slight preference for importing this near the end of the list, and keeping more module-specific and functional imports higher up.

File: apps/src/music/blockly/blocks/events.js [hunk 4]
```diff
@@ -49,7 +50,7 @@ export const triggeredAt = {
 export const triggeredAtSimple = {
   definition: {
     type: BlockTypes.TRIGGERED_AT_SIMPLE,
-    message0: '%1 triggered',
+    message0: `%1 ${musicI18n.blockly_blockTriggered()}`,
```
Comment (breville) [inline apps/src/music/blockly/blocks/events.js]: Should we incorporate the `%1` value into the loc'ed string, in case there is a translation where the `%1` value won't appear first?  Also it gives the translator more context.

Comment (breville) [reply apps/src/music/blockly/blocks/events.js]: Similar comment for a few similar cases below.

File: apps/i18n/music/en_us.json (hunk 1 above)
Comment (sanchitmalhotra126) [reply apps/i18n/music/en_us.json]: I went ahead and did this just to be safe, just in case we do end up pulling blocks from other modes into our current model

File: apps/src/music/blockly/blocks/control.js (hunk 2 above)
Comment (sanchitmalhotra126) [reply apps/src/music/blockly/blocks/control.js:6]: Yep, this gets handled by Blockly under the hood. Since this is nearly a direct copy of Blockly's for loop, I opted to use their translated string instead of handling the translation ourselves.

File: apps/src/music/blockly/blocks/events.js (hunk 4 above)
Comment (sanchitmalhotra126) [reply apps/src/music/blockly/blocks/events.js]: Yeah good question, wasn't sure about this. But I did end up doing this for a few strings that had `%1` in the middle of the string (ex. `play %1 at %2`), so it would be best to be consistent. I'll update this.

File: apps/src/music/blockly/fields.js [hunk 5]
```diff
@@ -59,10 +60,10 @@ export const fieldRestDurationDefinition = {
   type: 'field_dropdown',
   name: FIELD_REST_DURATION_NAME,
   options: [
-    ['\u00bd beat', '0.125'],
-    ['1 beat', '0.25'],
-    ['2 beats', '0.5'],
-    ['1 measure', '1'],
-    ['2 measures', '2'],
+    [`\u00bd ${musicI18n.blockly_fieldRestBeat()}`, '0.125'],
```
Comment (breville) [inline apps/src/music/blockly/fields.js]: Sorry for missing this earlier... I suspect the 1/4 character should also be part of the localized string, similar to the other numbers...

Comment (sanchitmalhotra126) [reply apps/src/music/blockly/fields.js]: Oh yes good call, the number characters should probably be localized too

Comment (sanchitmalhotra126) [reply apps/src/music/blockly/fields.js]: Updated to include this character and the number characters

**Review by breville — APPROVED** [review]
